### PR TITLE
Backwards Compatibility Tests Draft

### PR DESCRIPTION
**Pull request by peternied**
### Description
Using https://github.com/opensearch-project/anomaly-detection/pull/158 as a template.
    
Gradlew execution of tests has been implemented, but untested.  The
tests themselves have been started but do not have the actual security
plugin setup/validation logic that we would like to use.  Scan for TODO
within the files.

NOTE: Final commit [75bc096](https://github.com/opensearch-project/security/pull/1666/commits/75bc096f86ddc4231837aff1ccdb5a915ba3c83e) should be under consideration, all other changes were needed for the gradle downgrade, in https://github.com/opensearch-project/security/pull/1661

Approximant guess of
- 85% Gradle: Implemented, building, following the standards approach used by other repos, but as of yet untested.
- 40% Test case implementation, missing actual test setup/action/verification within the state of the test scenario, mostly security plugin specific knowledge needed at this point

### Issues Resolved
* Progress on https://github.com/opensearch-project/security/issues/1417

### Check List
- [ ] ~New functionality includes testing~
- [X] New functionality has been documented
- [X] Commits are signed per the DCO using --signoff

By submitting this pull request, I confirm that my contribution is made under the terms of the Apache 2.0 license.
For more information on following Developer Certificate of Origin and signing off your commits, please check [here](https://github.com/opensearch-project/OpenSearch/blob/main/CONTRIBUTING.md#developer-certificate-of-origin).

### CODE REVIEW

**Comment by peternied** [comment]
NOTE: This pull request is not indented to be merged, its but serves as a starting point for taking these tests across the finish line.

**Comment by cliu123** [comment]
> created those changes as a separate gradle file

@peternied Looks like this PR is changing the existing `build.gradle` file.

File: build.gradle [hunk 1]
```diff
@@ -374,4 +376,143 @@ buildRpm {
 buildDeb {
     arch = 'all'
     archiveName "${packageName}-${version}.deb"
+}
+
+String bwcVersion = "1.13.1.0";
+String baseName = "securityBwcCluster"
+String bwcFilePath = "src/test/resources/bwc"
+
+2.times {i ->
```
Comment (DarshitChanpura) [inline build.gradle:385]: can you add a comment here please explaining why we need two clusters here (just like you did below)

**Comment by peternied** [comment]
@cliu123 Thanks for finishing this work in https://github.com/opensearch-project/security/pull/1685

**Pull request closed without merging**